frc46_token/src/token/mod.rs

```diff
@@ -76,11 +77,24 @@ where
         Self { bs, msg, granularity, state }
     }
 
+    /// Replace the current state with another
+    /// The previous state is returned and can be safely dropped
+    pub fn replace(&mut self, state: TokenState) -> TokenState {
+        std::mem::replace(self.state, state)
```

Comment:
I don't think this is what I want. I think I was confused when I asked for it.

What I need would be be parallel with `wrap`, taking a mutable reference. I have a new `State` (embedded in another struct) that I've loaded into memory, and now want a token wrapping it. I want future mutations to mutate that reference in place (so I can persist later). I don't want to further mutate the state behind the old reference.

Perhaps I could just `wrap` again to get a new token in this case, but a `re-wrap(&mut new_state)` would still be convenient.

We should probably restore the old `replace` method for its use in testing.

Reply:
What you're describing might be better done with `std::mem::swap`. Though it may also be better to just treat the `Token` as disposable and create a new one as needed. Which I guess is what we'd end up doing in a re-wrap function - create a new Token by taking everything but the state from the old one.

This new `replace` doesn't hurt the tests, it's a better version of what they were using before.